platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolMain.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.demo.addressbook;
+
+import static com.swirlds.logging.LogMarker.STARTUP;
+
+import com.swirlds.common.system.BasicSoftwareVersion;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.system.Platform;
+import com.swirlds.common.system.PlatformWithDeprecatedMethods;
+import com.swirlds.common.system.SwirldMain;
+import com.swirlds.common.system.SwirldState;
+import java.security.SecureRandom;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * <p>
+ * An application that updates address book stake weights on version upgrade.
+ * </p>
+ *
+ * <p>
+ * Arguments:
+ * <ol>
+ * <li>
+ * No arguments parsed at this time.  The software version must be updated through setting the static value in
+ * this main class and recompiling. The behavior of staking is updated in the State class and recompiling.
+ * </li>
+ * </ol>
+ */
+public class AddressBookTestingToolMain implements SwirldMain {
+    private static final Logger logger = LogManager.getLogger(AddressBookTestingToolMain.class);
+
+    private static BasicSoftwareVersion softwareVersion = new BasicSoftwareVersion(1);
```

Comment:
For automated tests we won't want to hard code this. But I'm assuming you are saving that for a follow up PR after you add the new workflow for supporting application configuration using the new configuration engine.

Reply:
Once the Main can add its configuration in the browser, I will add the code to set the software version through querying the configuration value.